src/plan_expand_hypertable.c

```diff
@@ -138,9 +139,9 @@ collect_quals_mutator(Node *node, CollectQualCtx *ctx)
 								 "first parameter for chunks_in function needs to be a record")));
 
 				ctx->chunk_exclusion_func = func_expr;
+				f->quals = (Node *) list_delete_cell((List *) (f->quals), lc, prev);
 			}
-
-			if (ctx->chunk_exclusion_func == NULL)
+			else
```

Comment:
I think this change, changes the semantics in a bad way. Before the branch below was never run after the chunk exclusion function was found (in any clause). Now the brach is run unless the chunk_exclusion function is in the same clause. AFAIK the previous semantics is the right one.

Reply:
i'm not sure whats the right semantic here, but i will revert changes to that file, also this semantic is not covered by any tests